src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -193,15 +201,21 @@ private Object readResolve() {
     }
 
     private void setupTransients() {
+        runtimeThreads = new ConcurrentSkipListMap<>();
         runner = new CpsVmExecutorService(this);
         pausedByQuietMode = new AtomicBoolean();
     }
 
+    private synchronized Object writeReplace() {
```

Comment:
I think just

```suggestion
    private Object writeReplace() {
```

suffices